e2xgrader/apps/api.py

```diff
@@ -140,3 +214,91 @@ def get_task_submissions(self, assignment_id, notebook_id, task_id):
             submission['index'] = idx
 
         return submissions
+
+    def list_updated_cells(self, notebook: str, assignment: str) -> dict:
+        """Lists the updated autograde cell id and content from the source directory.
+
+        Arguments
+        ---------
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        Returns
+        -------
+        updated_cells: list
+            Dictionary where the keys are the ids of the cell and the values are the content.
+        """
+        nb = nbformat.read(os.path.join(self.coursedir.source_directory, assignment, notebook + '.ipynb'),
+                           as_version = nbformat.NO_CONVERT)
+        source_directory = {}
+        for cell in nb.cells:
```

Comment:
I think it might be nicer to implement this method like this:

```
from e2xgrader.utils.nbgrader_cells import grade_id
from nbgrader.utils import is_grade, is_solution, compute_checksum

.
.
.

nb_test_cells = {grade_id(cell): cell for cell in nb.cells 
                 if is_grade(cell) and not is_solution(cell)}

updated_notebook = self.gradebook.find_notebook(notebook, assignment)
source_cells = updated_notebook.source_cells

# The next three lines could be replaced by a call to list_autograde_testcells(...)
grade_cell_names = [cell.name for cell in updated_notebook.grade_cells]
solution_cell_names = [cell.name for cell in updated_notebook.solution_cells]
test_cell_names = set(grade_cell_names).difference(set(solution_cell_names))

gb_test_cells = [cell for cell in source_cells if cell.name in test_cell_names]
updated_cells = []
for cell in gb_test_cells:
    if cell.name in nb_test_cells and compute_checksum(nb_test_cells[cell.name]) != cell.checksum:
        updated_cells.append(cell.name)
return updated_cells
```